stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/MessageListFragment.kt

```diff
@@ -0,0 +1,229 @@
+import io.getstream.chat.android.ui.message.list.header.viewmodel.bindView
+import io.getstream.chat.android.ui.message.list.viewmodel.bindView
+import io.getstream.chat.android.ui.message.list.viewmodel.factory.MessageListViewModelFactory
+
+/**
+ * Self-contained chat screen which internally contains the following components:
+ *
+ * - [MessageListHeaderView] - displays the navigation icon, the channel information
+ *   and the channel image
+ * - [MessageListView] - shows a list of paginated messages, with threads, replies,
+ *   quotes, reactions and deleted messages
+ * - [MessageInputView] - allows the user to send new messages as well as pick and
+ *   choose attachments to send
+ *
+ * **Note**: Fragments representing self-contained screens are easy to use. They allow you
+ * to explore the SDK's features in a breeze, however, they offer limited customization.
+ */
+public open class MessageListFragment : Fragment() {
+
+    private val cid: String by lazy {
+        requireNotNull(requireArguments().getString(ARG_CHANNEL_ID)) { "Channel cid must not be null" }
+    }
+    private val themeResId: Int by lazy { requireArguments().getInt(ARG_THEME_RES_ID) }
+    private val messageId: String? by lazy { requireArguments().getString(ARG_MESSAGE_ID) }
+    private val showHeader: Boolean by lazy { requireArguments().getBoolean(ARG_SHOW_HEADER, false) }
+
+    private val factory: MessageListViewModelFactory by lazy { MessageListViewModelFactory(cid, messageId) }
+    private val messageListViewModel: MessageListViewModel by viewModels { factory }
+    private val messageListHeaderViewModel: MessageListHeaderViewModel by viewModels { factory }
+    private val messageInputViewModel: MessageInputViewModel by viewModels { factory }
+
+    private var backPressListener: BackPressListener? = null
+
+    private var _binding: StreamUiFragmentMessageListBinding? = null
+    protected val binding: StreamUiFragmentMessageListBinding get() = _binding!!
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        backPressListener = findListener()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        val layoutInflater = if (getTheme() != 0) {
+            inflater.cloneInContext(ContextThemeWrapper(context, getTheme()))
+        } else {
+            inflater
+        }
+        return StreamUiFragmentMessageListBinding.inflate(layoutInflater, container, false)
+            .apply { _binding = this }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupMessageListHeader()
+        setupMessageList()
+        setupMessageInput()
+    }
+
+    @StyleRes
+    protected open fun getTheme(): Int {
+        return themeResId
+    }
+
+    protected open fun setupMessageListHeader() {
```

Comment:
Really great docs on a lot of the declarations in this PR - can we please add docs on all of these `open` functions too?